diff --git a/node_modules/@hinkal/common/webworker/viteWorkerURL.constant.mjs b/node_modules/@hinkal/common/webworker/viteWorkerURL.constant.mjs
index ec5bc33..6d6451c 100644
--- a/node_modules/@hinkal/common/webworker/viteWorkerURL.constant.mjs
+++ b/node_modules/@hinkal/common/webworker/viteWorkerURL.constant.mjs
@@ -1,21 +1,9 @@
 
-  const domain = 'https://d13x2m0ev3hbsi.cloudfront.net';
-
-  // Returns a blob:// URL which points
-  // to a javascript file which will call
-  // importScripts with the given URL
-  async function getWorkerURL(url) {
-    const contentRes = await fetch(url);
-    const content = await contentRes.text()
-
-    return URL.createObjectURL(new Blob([content], { type: "application/javascript" }));
-  }
-
   var e = /* @__PURE__ */ ((r) => (r.ZKProof = "ZKProof", r.SnarkJS = "SnarkJS", r.UTXO = "UTXO", r))(e || {});
 
   const n = async () => ({
-    [e.ZKProof]: await getWorkerURL(domain + '/0.2.19/' + 'zkProofWorkerLauncher.js'),
-    [e.SnarkJS]: await getWorkerURL(domain + '/0.2.19/' + 'snarkjsWorkerLauncher.js'),
-    [e.UTXO]: await getWorkerURL(domain + '/0.2.19/' + 'utxoWorkerLauncher.js'),
+    [e.ZKProof]: '/assets/hinkal/zkProofWorkerLauncher.js',
+    [e.SnarkJS]: '/assets/hinkal/snarkjsWorkerLauncher.js',
+    [e.UTXO]:'/assets/hinkal/utxoWorkerLauncher.js',
   });
   export { n as getWorkerViteURL };
