diff --git a/node_modules/@hinkal/common/data-structures/snapshot/evm-snapshots/ClientAccessTokenSnapshotService.mjs b/node_modules/@hinkal/common/data-structures/snapshot/evm-snapshots/ClientAccessTokenSnapshotService.mjs
index b5d02f5..4f5cb74 100644
--- a/node_modules/@hinkal/common/data-structures/snapshot/evm-snapshots/ClientAccessTokenSnapshotService.mjs
+++ b/node_modules/@hinkal/common/data-structures/snapshot/evm-snapshots/ClientAccessTokenSnapshotService.mjs
@@ -3,6 +3,8 @@ import { getProviderFromContract as t } from "../../../functions/utils/getSigner
 import { AbstractAccessTokenSnapshotService as n } from "../../event-service/evm/AbstractAccessTokenSnapshotService.mjs";
 import "../../event-service/index.mjs";
 import { SnapshotFetcherService as r } from "../SnapshotFetcherService.mjs";
+import { get, set } from "idb-keyval";
+
 //#region libs/shared/common/src/data-structures/snapshot/evm-snapshots/ClientAccessTokenSnapshotService.ts
 var i = class extends n {
 	snapshotsService;
@@ -12,8 +14,21 @@ var i = class extends n {
 	async fetchSnapshot() {
 		let { contract: n } = this.blockchainEventEmitter, { chainId: i } = await t(n).getNetwork(), a = Number(i);
 		this.snapshotsService ||= new r(a, e(n));
+		const storedSnapshot = await get(`access_tokens_${i}`);
+     if(storedSnapshot) return storedSnapshot
+
 		let o = await this.snapshotsService.getAccessTokens();
 		if (o.accessTokenContractAddress !== e(n) || o.chainId !== a) throw Error("Commitment Snapshot: incorrect contract or chain id");
+
+		try{
+			     await set(`access_tokens_${i}`,{
+			        latestBlockNumber: o.latestBlockNumber,
+			        merkleTree: o.merkleTree,
+			       senderAddresses: o.senderAddresses,
+			        senderAddressIndexMap: o.senderAddressIndexMap
+			      })
+		    }catch{}
+
 		return {
 			latestBlockNumber: o.latestBlockNumber,
 			merkleTree: o.merkleTree,
@@ -21,8 +36,10 @@ var i = class extends n {
 			senderAddressIndexMap: o.senderAddressIndexMap
 		};
 	}
-	persistSnapshot(e) {
-		return Promise.resolve();
+	async persistSnapshot(e) {
+		try{
+		   await set(`access_tokens_${this.snapshotsService.chainId}`,e)
+	   }catch{}
 	}
 };
 //#endregion
diff --git a/node_modules/@hinkal/common/data-structures/snapshot/evm-snapshots/ClientApprovalsSnapshotService.mjs b/node_modules/@hinkal/common/data-structures/snapshot/evm-snapshots/ClientApprovalsSnapshotService.mjs
index e0cfd8e..2e91f1b 100644
--- a/node_modules/@hinkal/common/data-structures/snapshot/evm-snapshots/ClientApprovalsSnapshotService.mjs
+++ b/node_modules/@hinkal/common/data-structures/snapshot/evm-snapshots/ClientApprovalsSnapshotService.mjs
@@ -2,6 +2,8 @@ import { getContractAddress as e } from "../../../functions/utils/getContractAdd
 import { getProviderFromContract as t } from "../../../functions/utils/getSignerFromContract.mjs";
 import { AbstractApprovalsSnapshotService as n } from "../../event-service/evm/AbstractApprovalsSnapshotService.mjs";
 import "../../event-service/index.mjs";
+import { get, set } from "idb-keyval";
+
 //#region libs/shared/common/src/data-structures/snapshot/evm-snapshots/ClientApprovalsSnapshotService.ts
 var r = class extends n {
 	snapshotsService;
@@ -9,21 +11,45 @@ var r = class extends n {
 		super(e), this.snapshotsService = t;
 	}
 	async fetchSnapshot() {
-		let { contract: n } = this.blockchainEventEmitter, { chainId: r } = await t(n).getNetwork(), i = Number(r), a = await this.snapshotsService.getApprovals();
+	const { contract: n } = this.blockchainEventEmitter
+    const { chainId: r } = await t(n).getNetwork();
+    const storedSnapshot = await get(`approvals_${r}`);
+
+    if(storedSnapshot){
+      return {
+       latestBlockNumber: storedSnapshot.latestBlockNumber,
+        approvals: storedSnapshot.approvals,
+        events: storedSnapshot.events
+      }
+    }
+
+	const i = Number(r), a = await this.snapshotsService.getApprovals();
+
 		if (!a) return {
 			latestBlockNumber: 0,
 			approvals: {},
 			events: []
 		};
 		if (a.hinkalAddress !== e(n) || a.chainId !== i) throw Error("approvalsSnapshotData Snapshot: incorrect contract or chain id");
+
+		try{
+			  await set(`approvals_${r}`,{
+			        latestBlockNumber: a.latestBlockNumber,
+			        approvals: a.approvals,
+			        events: a.events
+			      })
+			    }catch{}
+
 		return {
 			latestBlockNumber: a.latestBlockNumber,
 			approvals: a.approvals,
 			events: a.events
 		};
 	}
-	persistSnapshot(e) {
-		return Promise.resolve();
+	async persistSnapshot(e) {
+		try{
+			 await set(`approvals_${this.snapshotsService.chainId}`,e)
+		  }catch{}
 	}
 };
 //#endregion
diff --git a/node_modules/@hinkal/common/data-structures/snapshot/evm-snapshots/ClientCommitmentsSnapshotService.mjs b/node_modules/@hinkal/common/data-structures/snapshot/evm-snapshots/ClientCommitmentsSnapshotService.mjs
index e4c90b3..f523538 100644
--- a/node_modules/@hinkal/common/data-structures/snapshot/evm-snapshots/ClientCommitmentsSnapshotService.mjs
+++ b/node_modules/@hinkal/common/data-structures/snapshot/evm-snapshots/ClientCommitmentsSnapshotService.mjs
@@ -1,5 +1,7 @@
 import { AbstractCommitmentsSnapshotService as e } from "../../event-service/evm/AbstractCommitmentsSnapshotService.mjs";
 import "../../event-service/index.mjs";
+import { get, set } from "idb-keyval";
+
 //#region libs/shared/common/src/data-structures/snapshot/evm-snapshots/ClientCommitmentsSnapshotService.ts
 var t = class extends e {
 	snapshotsService;
@@ -7,7 +9,22 @@ var t = class extends e {
 		super(e), this.snapshotsService = t;
 	}
 	async fetchSnapshot() {
-		let e = await this.snapshotsService.getCommitments();
+		const storedSnapshot = await get(`commitments_${this.snapshotsService.chainId}`);
+		
+		    if(storedSnapshot) return storedSnapshot;
+
+			let e = await this.snapshotsService.getCommitments();
+		
+		    try{
+		      await set(`commitments_${this.snapshotsService.chainId}`,{
+		        latestBlockNumber: e.latestBlockNumber,
+		        merkleTree: e.merkleTree,
+		        encryptedOutputs: e.encryptedOutputs,
+				encryptedOutputsByCommitment: e.encryptedOutputsByCommitment
+		      });
+		    }catch{}
+		
+
 		return {
 			latestBlockNumber: e.latestBlockNumber,
 			merkleTree: e.merkleTree,
@@ -15,8 +32,10 @@ var t = class extends e {
 			encryptedOutputsByCommitment: e.encryptedOutputsByCommitment
 		};
 	}
-	persistSnapshot(e) {
-		return Promise.resolve();
+	async persistSnapshot(e) {
+		try{
+			await set(`commitments_${this.snapshotsService.chainId}`,e)
+		}catch{}
 	}
 };
 //#endregion
diff --git a/node_modules/@hinkal/common/data-structures/snapshot/evm-snapshots/ClientNullifierSnapshotService.mjs b/node_modules/@hinkal/common/data-structures/snapshot/evm-snapshots/ClientNullifierSnapshotService.mjs
index 71cf5e2..43ddd47 100644
--- a/node_modules/@hinkal/common/data-structures/snapshot/evm-snapshots/ClientNullifierSnapshotService.mjs
+++ b/node_modules/@hinkal/common/data-structures/snapshot/evm-snapshots/ClientNullifierSnapshotService.mjs
@@ -1,5 +1,7 @@
 import { AbstractNullifierSnapshotService as e } from "../../event-service/evm/AbstractNullifierSnapshotService.mjs";
 import "../../event-service/index.mjs";
+import { get, set } from "idb-keyval";
+
 //#region libs/shared/common/src/data-structures/snapshot/evm-snapshots/ClientNullifierSnapshotService.ts
 var t = class extends e {
 	snapshotsService;
@@ -7,14 +9,33 @@ var t = class extends e {
 		super(e), this.snapshotsService = t;
 	}
 	async fetchSnapshot() {
+		const storedSnapshot = await get(`nullifiers_${this.snapshotsService.chainId}`);
+		
+		    if(storedSnapshot){
+		      return {
+		        latestBlockNumber: storedSnapshot.latestBlockNumber,
+		        nullifiers: storedSnapshot.nullifiers
+		      }
+		    }
+
 		let e = await this.snapshotsService.getNullifiers();
+
+		try{
+			    await set(`nullifiers_${this.snapshotsService.chainId}`,{
+			        latestBlockNumber: e.latestBlockNumber,
+			        nullifiers: e.nullifiers
+			      })
+		    }catch{}
+
 		return {
 			latestBlockNumber: e.latestBlockNumber,
 			nullifiers: e.nullifiers
 		};
 	}
-	persistSnapshot(e) {
-		return Promise.resolve();
+	async persistSnapshot(e) {
+		try{
+			    await set(`nullifiers_${this.snapshotsService.chainId}`,e)
+		    }catch{}
 	}
 };
 //#endregion
diff --git a/node_modules/@hinkal/common/webworker/viteWorkerURL.constant.mjs b/node_modules/@hinkal/common/webworker/viteWorkerURL.constant.mjs
index db1d1c5..d7be300 100644
--- a/node_modules/@hinkal/common/webworker/viteWorkerURL.constant.mjs
+++ b/node_modules/@hinkal/common/webworker/viteWorkerURL.constant.mjs
@@ -1,21 +1,8 @@
-
-  const domain = 'https://d13x2m0ev3hbsi.cloudfront.net';
-
-  // Returns a blob:// URL which points
-  // to a javascript file which will call
-  // importScripts with the given URL
-  async function getWorkerURL(url) {
-    const contentRes = await fetch(url);
-    const content = await contentRes.text()
-
-    return URL.createObjectURL(new Blob([content], { type: "application/javascript" }));
-  }
-
-  var e = /* @__PURE__ */ ((r) => (r.ZKProof = "ZKProof", r.SnarkJS = "SnarkJS", r.UTXO = "UTXO", r))(e || {});
+var e = /* @__PURE__ */ ((r) => (r.ZKProof = "ZKProof", r.SnarkJS = "SnarkJS", r.UTXO = "UTXO", r))(e || {});
 
   const n = async () => ({
-    [e.ZKProof]: await getWorkerURL(domain + '/0.2.21/' + 'zkProofWorkerLauncher.js'),
-    [e.SnarkJS]: await getWorkerURL(domain + '/0.2.21/' + 'snarkjsWorkerLauncher.js'),
-    [e.UTXO]: await getWorkerURL(domain + '/0.2.21/' + 'utxoWorkerLauncher.js'),
+    [e.ZKProof]: '/assets/hinkal/zkProofWorkerLauncher.js',
+    [e.SnarkJS]: '/assets/hinkal/snarkjsWorkerLauncher.js',
+    [e.UTXO]: '/assets/hinkal/utxoWorkerLauncher.js'
   });
   export { n as getWorkerViteURL };
